### PR TITLE
Allow NettyPerfClient to connect multiple frontends

### PR DESCRIPTION
**Pull request by jsjtzyy**
1. log response based on response code range rather than single response
code value.
2. allow perfClient to connect multiple frontends.

### CODE REVIEW

File: ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java [hunk 1]
```diff
@@ -446,8 +462,15 @@ public void channelRead0(ChannelHandlerContext ctx, HttpObject in) {
...
-          logger.error("Got Response code {} and headers were {}", response.status().code(), response.headers());
+        if (response.status().code() >= 200 && response.status().code() < 300) {
+          logger.trace("Request succeeded");
+          if (response.headers().contains("Location")) {
+            logger.info(response.headers().get("Location"));
+          }
+        } else if (response.status().code() >= 300 && response.status().code() < 400) {
+          logger.warn("Redirection code {} and headers were {}", response.status().code(), response.headers());
+        } else {
+          logger.error("Response error code {} and headers were {}", response.status().code(), response.headers());
```
Comment (justinlin-linkedin) [inline ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: any reason to log these messages?

File: ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java [hunk 2]
```diff
@@ -355,20 +360,30 @@ protected void start() {
     b.group(group).channel(NioSocketChannel.class).handler(new ChannelInitializer<SocketChannel>() {
       @Override
       public void initChannel(SocketChannel ch) throws Exception {
+        channelConnectLatch.await();
         if (sslFactory != null) {
+          String host = channelToHost.get(ch);
+          if (host == null) {
+            throw new Exception("Unrecognized host!");
+          }
+          logger.info("Initializing the channel to {}", host);
```
Comment (justinlin-linkedin) [inline ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: this if statement and logging could be moved to the chnnalConnectListener so we can get rid of the latch wait here.

File: ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java (hunk 1 above)
Comment (jsjtzyy) [reply ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: Recording 50x error code might be useful to remind us there is something wrong on server side. 40x can drive us to double check if server behaves as expected.

File: ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java (hunk 2 above)
Comment (justinlin-linkedin) [reply ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: resolved by offline discussion

File: ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java [hunk 3]
```diff
@@ -136,11 +140,12 @@
      * Parses the arguments provided and extracts them into variables that can be retrieved.
      * @param args the command line argument list.
      */
-    protected ClientArgs(String args[]) {
+    ClientArgs(String[] args) {
```
Comment (zzmao) [inline ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: Minor: Use 'protected' to be more strict?

Comment (zzmao) [reply ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: Please ignore this. Looks good.

**Pull request merged**